### ISSUE TITLE
Disabling API access from notebook container.

### ISSUE THREAD

**Issue by GrahamDumpleton** [jupyterhub/kubespawner#109]
Instead of:

```
      volumeMounts:
        - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
          name: no-api-access-please
          readOnly: true
```

to block access to service account token from within a container, why not use ``automountServiceAccountToken``?

* https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/

From 1.6 you can define it on a ``Pod``.

Do you still need to support pre 1.6?

**Comment by yuvipanda** [comment]
I think we can remove pre 1.6 support next version of kubespawner! If we do, I'd like us to make sure that we fail *very* explicitly if it's not supported, since the failure more of 'oops, now everyone has access to your k8s api!' is pretty bad.

**Comment by GrahamDumpleton** [comment]
If only people used RBAC. :-)

In OpenShift the 'default' service account that applications run under by default, although the access token is mounted in the container, cannot actually access the REST API and will be blocked. Unless of course someone mucks up RBAC on their project and grants all service accounts access to their project.

**Comment by consideRatio** [comment]
Ah - Weird mount understood better =)

**Comment by consideRatio** [comment]
Closed by #123
